backend/graph/api/schema.resolvers.go

```diff
@@ -250,6 +252,76 @@ func (r *mutationRootResolver) UpdateTaskMessage(ctx context.Context, id string,
 	return id, err
 }
 
+// SendEpisodeFeedback is the resolver for the sendEpisodeFeedback field.
+func (r *mutationRootResolver) SendEpisodeFeedback(ctx context.Context, episodeID string, message *string, rating *int) (string, error) {
+	_, err := getProfile(ctx)
+	if err != nil {
+		return "", err
+	}
+	episode, err := getEpisode(ctx, r.Resolver, episodeID)
+	if err != nil {
+		return "", err
+	}
+	err = ratelimit.Endpoint(ctx, "episodes:messages:send:"+strconv.Itoa(episode.ID), 2, false)
+	if err != nil {
+		return "", err
+	}
+	id, err := utils.GenerateRandomSecureString(32)
+	if err != nil {
+		return "", err
+	}
+	var str string
+	if message != nil {
+		str = *message
+	}
+	var metadata pqtype.NullRawMessage
+	if rating != nil {
+		metadata.RawMessage = []byte(fmt.Sprintf("{\"rating\": %d}", rating))
+		metadata.Valid = true
+	}
+	err = r.Queries.SetMessage(ctx, sqlc.SetMessageParams{
+		ID:       id,
+		Message:  str,
+		ItemID:   episode.UUID,
+		Metadata: metadata,
+	})
+	if err != nil {
+		log.L.Error().Err(err).Msg("Failed to save string to database")
+		return "", merry.New("Failed to generate unique ID")
+	}
+	return id, nil
+}
+
+// UpdateEpisodeFeedback is the resolver for the updateEpisodeFeedback field.
+func (r *mutationRootResolver) UpdateEpisodeFeedback(ctx context.Context, id string, message *string, rating *int) (string, error) {
+	_, err := getProfile(ctx)
+	if err != nil {
+		return "", err
+	}
+	err = ratelimit.Endpoint(ctx, "episodes:messages:update", 100, false)
+	if err != nil {
+		return "", err
+	}
+	var metadata pqtype.NullRawMessage
+	if rating != nil {
+		metadata.RawMessage = []byte(fmt.Sprintf("{\"rating\": %d}", rating))
```

Comment:
I know it's annoying but can we do a struct and marshall it? :D
Can use `gin.H` (https://github.com/gin-gonic/gin/blob/v1.8.1/utils.go#L54) which is basically a dict. Something like:

```golang
		metadata.RawMessage = json.Marshall(gin.H{"rating", rating}}
```

Reply:
I don't disagree, and I cleaned up a bit more.

Map is fine for this right?

Reply:
Sounds good to me.